meson_options.txt

```diff
@@ -0,0 +1,6 @@
+option('build-date', type: 'string', description: 'build date. Defaults to today if not set')
+option('enable-compat-getopt', type: 'boolean', value: false, description: 'enable getopt compatibility support')
```

Comment:
It is possible to autodetect it?

Reply:
https://github.com/pciutils/pciutils/blob/db43fb5e8f2c04c409bdd06ac2c2828685038d69/README.Windows#L30-L31

Meson could autodetect this by compiling a test program and seeing if it works. Under what circumstances is the compat-getopt needed? Is it enough to check if getopt is available in the libc?

Reply:
It is needed when libc/system does not provide one, mostly for older mingw32 runtime.

Reply:
> It is possible to autodetect it?

updated to auto-detect if compatibility getopt is needed.